src/engine/renderer/tr_shader.cpp

```diff
@@ -1404,7 +1404,7 @@ static bool ParseMap( const char **text, char *buffer, int bufferSize )
 	return true;
 }
 
-static bool LoadMap( shaderStage_t *stage, const char *buffer, const int bundleIndex = TB_COLORMAP )
+static bool LoadMap( shaderStage_t *stage, const char *buffer, stageType_t type, const int bundleIndex = TB_COLORMAP )
```

Comment:
In which case is `type` meaningfully different from `stage->type`? (I saw one case where `stage->type` is ATTENUATIONMAP_Z and `type` is COLORMAP but this doesn't matter as neither receives special treatment.)

Reply:
With a legacy material like this:

```
textures/blah
{
    diffuseMap blah_d
    normalMap blah_n
    specularMap blah_s
}
```

or like this (exactly the same but more verbose):

```
textures/blah
{
  {
    map blah_d
    stage diffuseMap
  }
  {
    map blah_n
    stage normalMap
  }
  {
    map blah_s
    stage specularMap
  }
}
```

Every stage is an actual stage and every map (image) is a different stage.

But with a material with a pre-collapsed stage like this

```
textures/blah
{
  {
    diffuseMap blah_d
    normalMap blah_n
    specularMap blah_s
  }
}
```

There is a single stage, for which the type is `diffuseMap`, but three maps (images), for which each type is `diffuseMap`, `normalMap` and `specularMap`.

When loading a map (image), we need to know for what kind of stage it is, being already collapsed or not.

With this pre-collapsed stage, the stage type is always `diffuseMap`, but we tell the image loader to run the specific code for `normalMap` stage when loading the normal maps and the one for `specularMap` stage when loading the specular map.

Passing an explicit `stageType_t` for loading an image that may differ from the stage type itself (because of multitexturing) was the least intrusive way, alternative would be to have an `imageType_t` which would require much more development, and would be prone to mistakes since we would always have to keep it in sync with `stageType_t` anyway.

Reply:
Actually the reuse of those `stageType_t` for image types is the consequence of the fact we not only need to have map types, but to duplicate map types as stage types to keep compatibility with legacy syntaxes. And since all those image types also have to be separate stage types before collapsing, I just reuse them for image types. But in fact those stage types are the reuse of image types, it's just less verbose, less prone to errors, less messy and less intrusive to have a single enum for everything.